### PR TITLE
Validator

### PR DESCRIPTION
**Pull request by YuvNavon**
@idangrosbard

### CODE REVIEW

**Comment by idangrosbard** [comment]
1. Add `self` parameter to all function definitions
2. Add python typing for all function inputs and outputs
3. Add pytests for good and bad input cases. Check that each validator function works well

**Review by idangrosbard — CHANGES_REQUESTED** [review]
Add python typing
Add `self` parameter to all functions
Add pytests for all functions

File: validator.py [hunk 1]
```diff
@@ -63,25 +69,63 @@ def check_coordinates_range(list_of_coordinates):
...
+
+def test_check_coordinates_is_not_empty(validator):
+    assert validator.check_coordinates_is_not_empty([[1, 2, 3], [4, 5, 6]])
+    assert not validator.check_coordinates_is_not_empty([])
+
+def test_check_coordinates_range(validator):
+    assert validator.check_coordinates_range([[1, 2, 3], [4, 5, 6]])
+    assert not validator.check_coordinates_range([[100, 2, 3], [4, 5, 6]])
+
+def test_test_coordinates(validator):
```
Comment (ORachelS) [inline validator.py:123]: might be redundant, we already checked each one individually

File: validator.py [hunk 2]
```diff
@@ -63,25 +69,63 @@ def check_coordinates_range(list_of_coordinates):
...
         return True
     
 
+import pytest
+
+"""
+Tests for the Validator class
+"""
+
+def test_check_coordinates_is_int(validator):
```
Comment (ORachelS) [inline validator.py:103]: maybe add a message that the tests pass or failed (the test for the validations)

**Review by ORachelS — CHANGES_REQUESTED** [review]

**Review by ORachelS — APPROVED** [review]
looks great

**Review by ORachelS — APPROVED** [review]

**Review by idangrosbard — CHANGES_REQUESTED** [review]
Change `List[List[int]]` to `np.array` (numpy)

**Review by idangrosbard — APPROVED** [review]
Looks great :) 10 / 10

**Pull request merged**